File: app/src/main/java/com/rafaeltmbr/stopwatch/domain/data/repositories/impl/StopwatchRepositoryImpl.kt
package com.rafaeltmbr.stopwatch.domain.data.repositories.impl

import com.rafaeltmbr.stopwatch.domain.data.data_sources.StopwatchDataSource
import com.rafaeltmbr.stopwatch.domain.data.repositories.StopwatchRepository
import com.rafaeltmbr.stopwatch.domain.entities.StopwatchState


class StopwatchRepositoryImpl(
    private val dao: StopwatchDataSource
) : StopwatchRepository {
    override suspend fun load() = dao.load()

    override suspend fun save(state: StopwatchState) = dao.save(state)
}

File: ios/Stopwatch/core/data/repositories/impl/StopwatchRepositoryImpl.swift
struct StopwatchRepositoryImpl: StopwatchRepository {
    private let stopwatchDataSource: StopwatchDataSourcePort
    
    init(_ stopwatchDataSource: StopwatchDataSourcePort) {
        self.stopwatchDataSource = stopwatchDataSource
    }
    
    func load() async -> StopwatchState? {
        return await stopwatchDataSource.load()
    }
    
    func save(_ state: StopwatchState) async {
        await stopwatchDataSource.save(state)
    }
}
